src/current/v23.2/protect-changefeed-data.md

```diff
@@ -0,0 +1,79 @@
+---
+title: Protect Changefeed Data from Garbage Collection
+summary: Understand how changefeeds interact with garbage collection.
+toc: true
+docs_area: stream_data
+---
+
+By default, [protected timestamps]({% link {{ page.version.version }}/architecture/storage-layer.md %}#protected-timestamps) will protect changefeed data from [garbage collection]({% link {{ page.version.version }}/architecture/storage-layer.md %}#garbage-collection) up to the time of the [_checkpoint_]({% link {{ page.version.version }}/how-does-an-enterprise-changefeed-work.md %}).
+
+Protected timestamps will protect changefeed data from garbage collection in the following scenarios:
+
+- The downstream [changefeed sink]({% link {{ page.version.version }}/changefeed-sinks.md %}) is unavailable. Protected timestamps will protect changes until you either [cancel]({% link {{ page.version.version }}/cancel-job.md %}) the changefeed or the sink becomes available once again.
+- (**deprecated**) You [pause]({% link {{ page.version.version }}/pause-job.md %}) a changefeed with the [`protect_data_from_gc_on_pause`]({% link {{ page.version.version }}/create-changefeed.md %}#protect-pause) option enabled. Or, a changefeed with `protect_data_from_gc_on_pause` pauses from a [retryable error]({% link {{ page.version.version }}/monitor-and-debug-changefeeds.md %}#changefeed-retry-errors). Protected timestamps will protect changes until you [resume]({% link {{ page.version.version }}/resume-job.md %}) the changefeed.
+
+However, if the changefeed lags too far behind, the protected changes could lead to an accumulation of garbage. This could result in increased disk usage and degraded performance for some workloads.
+
+## Prevent garbage accumulation
+
+To prevent an accumulation of protected changes that could impact performance, consider the following:
+
+- [`changefeed.protect_timestamp.max_age`](#changefeed-protect_timestamp-max_age)
+- [`gc_protect_expires_after`](#gc_protect_expires_after)
+
+### `changefeed.protect_timestamp.max_age`
+
+{% include_cached new-in.html version="v23.2" %} By default, the `changefeed.protect_timestamp.max_age` [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}) sets the maximum time that changefeeds making no forward progress will hold protected timestamp records. Once the `changefeed.protect_timestamp.max_age` duration is reached, the changefeed will fail with a permanent error. As a result, it is **critical to monitor for changefeed failures** because changefeeds will eventually fail with an unrecoverable error if they cannot progress before the duration is reached.
+
+This cluster setting is enabled by default to 4 days.
```

Comment:
You can also mention (though this is in the setting doc string) -- that this functionality may be disabled by setting max_age to 0 (though, we recommend against that).

Reply:
Added.